File: testpaper/merge_acc.py
import re

with open('acc1_ch1_part1.js', 'r', encoding='utf-8') as f:
    part1 = f.read()
with open('acc1_ch1_part2.js', 'r', encoding='utf-8') as f:
    part2 = f.read()
with open('acc1_ch1_part3.js', 'r', encoding='utf-8') as f:
    part3 = f.read()

# Extract just the arrays from the files
def get_array_content(js_code):
    start = js_code.find('[')
    end = js_code.rfind(']')
    return js_code[start+1:end].strip()

arr1 = get_array_content(part1)
arr2 = get_array_content(part2)
arr3 = get_array_content(part3)

combined = '[\n' + arr1 + ',\n' + arr2 + ',\n' + arr3 + '\n            ]'

with open('data.js', 'r', encoding='utf-8') as f:
    data = f.read()

pattern = re.compile(r'(chapterName:\s*"Chapter 1 : Introduction to Accounting",\s*mcqData:\s*)\[[\s\S]*?\],\s*(shortCQData:)', re.MULTILINE)
data = pattern.sub(r'\1' + combined.replace('\\', '\\\\') + r',\n            \2', data)

with open('data.js', 'w', encoding='utf-8') as f:
    f.write(data)

print("Merged successfully.")

File: testpaper/merge_acc.js
const fs = require('fs');

const code1 = fs.readFileSync('acc1_ch1_part1.js', 'utf8');
const code2 = fs.readFileSync('acc1_ch1_part2.js', 'utf8');
const code3 = fs.readFileSync('acc1_ch1_part3.js', 'utf8');

let acc1_ch1_part1;
let acc1_ch1_part2;
let acc1_ch1_part3;

eval(code1);
eval(code2);
eval(code3);

const fullArray = [...acc1_ch1_part1, ...acc1_ch1_part2, ...acc1_ch1_part3];

// Let's modify data.js
let dataJs = fs.readFileSync('data.js', 'utf8');

// Format the array elements nicely to fit the data.js style
const formattedArray = "[\n" + fullArray.map(q => {
    return `                {\n                    id: ${q.id},\n                    text: ${JSON.stringify(q.text)},\n                    meta: ${JSON.stringify(q.meta)},\n                    options: ${JSON.stringify(q.options)},\n                    correctAnswer: ${q.correctAnswer}\n                }`;
}).join(",\n") + "\n            ]";

const regex = /(chapterName:\s*"Chapter 1 : Introduction to Accounting",\s*mcqData:\s*)\[[\s\S]*?\],\s*(shortCQData:)/;
dataJs = dataJs.replace(regex, `$1${formattedArray},\n            $2`);

fs.writeFileSync('data.js', dataJs);
console.log("Successfully merged " + fullArray.length + " questions into data.js");
